### PR TITLE
Export the contextHandler type

### PR DESCRIPTION
**Pull request by pmatseykanets**
to make it easier to extend siesta

### CODE REVIEW

File: tree.go [hunk 1]
```diff
@@ -389,6 +390,7 @@ walk: // Outer loop for walking the tree
 					}
 
 					if handle, usage = n.handle, n.usage; handle != nil {
+						fmt.Println(n.path, "1")
```
Comment (gkristic-vc) [inline tree.go:393]: Leftover?

File: tree.go [hunk 2]
```diff
@@ -412,6 +414,7 @@ walk: // Outer loop for walking the tree
 
 					handle = n.handle
 					usage = n.usage
+					fmt.Println(n.path, "2")
```
Comment (gkristic-vc) [inline tree.go:417]: Idem.

File: tree.go [hunk 3]
```diff
@@ -422,6 +425,7 @@ walk: // Outer loop for walking the tree
 			// We should have reached the node containing the handle.
 			// Check if this node has a handle registered.
 			if handle, usage = n.handle, n.usage; handle != nil {
+				fmt.Println(n.path, "3")
```
Comment (gkristic-vc) [inline tree.go:428]: Idem.

File: handler.go [hunk 4]
```diff
@@ -7,54 +7,55 @@ import (
...
+//     func(http.ResponseWriter, *http.Request, func())
+//     func(Context, http.ResponseWriter, *http.Request)
+//     func(Context, http.ResponseWriter, *http.Request, func())
+func ToContextHandler(f interface{}) ContextHandler {
 	switch f.(type) {
 	case func(Context, http.ResponseWriter, *http.Request, func()):
-		m = contextHandler(f.(func(Context, http.ResponseWriter, *http.Request, func())))
-	case contextHandler:
-		m = f.(contextHandler)
+		return ContextHandler(f.(func(Context, http.ResponseWriter, *http.Request, func())))
```
Comment (efepapa) [inline handler.go:30]: idiomatic `switch t := f.(type) {`?  That'd simplify most returns, this one would be `return ContextHandler(t)`

File: tree.go (hunk 1 above)
Comment (pmatseykanets) [reply tree.go]: 🤦‍♂️ 👍

**Pull request merged**